From 97a8b1616da0f9c62f11041716532e94cde137c3 Mon Sep 17 00:00:00 2001
From: Evgeniy Galyuta <olimpik.net@gmail.com>
Date: Tue, 3 Mar 2026 00:02:49 +0300
Subject: [PATCH] uniform names

---
 x265_encoder_10b.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x265_encoder_10b.cpp b/x265_encoder_10b.cpp
index 4edb378..a46bf2f 100644
--- a/x265_encoder_10b.cpp
+++ b/x265_encoder_10b.cpp
@@ -372,7 +372,7 @@ StatusCode X265Encoder10b::s_RegisterCodecs(HostListRef* p_pList)
 	const char* pCodecName = "Auto";
 	codecInfo.SetProperty(pIOPropName, propTypeString, pCodecName, static_cast<int>(strlen(pCodecName)));
 
-	const char* pCodecGroup = "X265 Main10";
+	const char* pCodecGroup = "X265 Main10 [x265-10b]";
 	codecInfo.SetProperty(pIOPropGroup, propTypeString, pCodecGroup, static_cast<int>(strlen(pCodecGroup)));
 
 	uint32_t vFourCC = 'hvc1';
-- 
2.51.2

