From 248662f461081a4b3767844e8fa14b5d9a8fa223 Mon Sep 17 00:00:00 2001
From: Evgeniy Galyuta <olimpik.net@gmail.com>
Date: Sun, 23 Aug 2026 21:40:59 +0300
Subject: [PATCH] uniform names

---
 aac_encoder.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aac_encoder.cpp b/aac_encoder.cpp
index 451db15..6a48572 100644
--- a/aac_encoder.cpp
+++ b/aac_encoder.cpp
@@ -112,7 +112,7 @@ StatusCode AACEncoder::s_RegisterCodecs(HostListRef* p_pList)
 
     codecInfo.SetProperty(pIOPropUUID, propTypeUInt8, AACEncoder::s_UUID, 16);
 
-    const char* pCodecName = "AAC 2.0 (FDK-AAC)";
+    const char* pCodecName = "AAC 2.0 [aac-fdk]";
     codecInfo.SetProperty(pIOPropName, propTypeString, pCodecName, strlen(pCodecName));
 
     uint32_t val = 'aac ';
-- 
2.54.0

